prestoadmin/util/presto_config.py

```diff
@@ -68,21 +68,17 @@ def from_file(config, config_path=None, config_host=None):
     @staticmethod
     def config(host, user):
         config_path = os.path.join(REMOTE_CONF_DIR, CONFIG_PROPERTIES)
-        try:
-            data = StringIO()
-            with settings(host_string='%s@%s' % (user, host)):
-                with hide('stderr', 'stdout'):
-                    temp_dir = run('mktemp -d /tmp/prestoadmin.XXXXXXXXXXXXXX')
-                try:
-                    get(config_path, data, use_sudo=True, temp_dir=temp_dir)
-                finally:
-                    run('rm -r %s' % temp_dir)
-
-            data.seek(0)
-            return PrestoConfig.from_file(data, config_path, host)
-        except Exception as e:
-            _LOGGER.info('Could not find Presto config.')
-            return PrestoConfig(None, config_path, host)
+        data = StringIO()
```

Comment:
1. instead of propagating  exceptions for things that could be user facing, you should use the error() function to give a nicer error message.  
2.  This was here because the client gets used for `server status`, and if a user calls `server status` without Presto installed, we don't want an exception.  We want to report back to the user that Presto is not installed. 
However, I think this change is a good one.  Status should be checking whether Presto is installed first, and not call out to the client if it knows that it's not (now the check happens somewhere in the middle of everything). The code for status is a bit messy, but for your change to work correctly, you would need to change `server.get_status_from_coordinator()` to check that Presto is installed before trying to query anything else.